### ISSUE TITLE
Allow integration of ssh-agent-based RS256 signature

### ISSUE THREAD

**Issue by mathieu-lacage** [jpadilla/pyjwt#639]
I'd like to be able to use pyjwt to generate jwt tokens signed by my ssh-agent. i.e., something like this: 

https://github.com/orion-labs/jwt-ssh-agent-go

I already implemented this with a python script that first generates a signing_input, sends it to connects to my local ssh-agent and completes the creation of the jwt by concatenating the signing_output to the signing_input.

Basically, the dumbest possible signing function:

```
def jwt_sign(key, claims):
     header_payload = json.dumps({"typ": 'JWT', "alg": 'RS256'}, separators=(",", ":")).encode()
     claims_payload = json.dumps(claims, separators=(",", ":")).encode("utf-8")
     segments = [
         base64.urlsafe_b64encode(header_payload),
         base64.urlsafe_b64encode(claims_payload)
     ]
     signing_input = b".".join(segments)
     signing_output = ssh_agent_sign(key, signing_input)
     segments.append(base64.urlsafe_b64encode(signing_output)) 
     return b".".join(segments)
```

I could live with this but I would also be happy to do the work to either:
1. push the code for this in pyjwt to allow pyjwt to transparently perform jwt generation and verification with an ssh-agent
2. find a way to register an RS256-compatible ssh-agent-based algorithm in pyjwt and publish that algorithm somewhere else if you do not like the idea of having the code for 1. in pyjwt

So, basically: are you interested in a patch for 1, or if not, could you suggest a way for me to do 2?
